baremetal/machine_manager_test.go

```diff
@@ -0,0 +1,1563 @@
+					Namespace: "myns",
+				},
+			},
+			ExpectedImage: &bmh.Image{
+				URL:      testImageURL,
+				Checksum: testImageChecksumURL,
+			},
+			ExpectUserData: true,
+		},
+
+		{
+			Scenario:                  "Externally provisioned, same machine",
+			UserDataNamespace:         "",
+			ExpectedUserDataNamespace: "myns",
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "host2",
+					Namespace: "myns",
+				},
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "machine1",
+						Namespace:  "myns",
+						Kind:       "Machine",
+						APIVersion: capi.GroupVersion.String(),
+					},
+				},
+			},
+			ExpectedImage: &bmh.Image{
+				URL:      testImageURL,
+				Checksum: testImageChecksumURL,
+			},
+			ExpectUserData: true,
+		},
+
+		{
+			Scenario:                  "Previously provisioned, different image, unchanged",
+			UserDataNamespace:         "",
+			ExpectedUserDataNamespace: "myns",
+			Host: bmh.BareMetalHost{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "host2",
+					Namespace: "myns",
+				},
+				Spec: bmh.BareMetalHostSpec{
+					ConsumerRef: &corev1.ObjectReference{
+						Name:       "machine1",
+						Namespace:  "myns",
+						Kind:       "Machine",
+						APIVersion: capi.GroupVersion.String(),
+					},
+					Image: &bmh.Image{
+						URL:      testImageURL + "test",
+						Checksum: testImageChecksumURL + "test",
+					},
+				},
+			},
+			ExpectedImage: &bmh.Image{
+				URL:      testImageURL + "test",
+				Checksum: testImageChecksumURL + "test",
+			},
+			ExpectUserData: false,
+		},
+	} {
+
+		t.Run(tc.Scenario, func(t *testing.T) {
+			// test data
+			bmmconfig, infrastructureRef := newConfig(t, tc.UserDataNamespace, map[string]string{}, []capbm.HostSelectorRequirement{})
+			machine := capi.Machine{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine1",
+					Namespace: "myns",
+				},
+				Spec: capi.MachineSpec{
+					InfrastructureRef: *infrastructureRef,
+				},
+			}
+
+			// test setup
+			scheme := runtime.NewScheme()
+			err := bmoapis.AddToScheme(scheme)
+
+			if err != nil {
+				log.Printf("AddToScheme failed: %v", err)
+			}
+
+			c := fakeclient.NewFakeClientWithScheme(scheme, &tc.Host)
+
+			actuator, err := newMachineManager(c, nil, nil, &machine, bmmconfig)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// run the function
+			err = actuator.setHostSpec(context.TODO(), &tc.Host)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// get the saved result
+			savedHost := bmh.BareMetalHost{}
+			err = c.Get(context.TODO(), client.ObjectKey{Name: tc.Host.Name, Namespace: tc.Host.Namespace}, &savedHost)
+			if err != nil {
+				t.Errorf("%v", err)
+				return
+			}
+
+			// validate the result
+			if savedHost.Spec.ConsumerRef == nil {
+				t.Errorf("ConsumerRef not set")
+				return
+			}
+			if savedHost.Spec.ConsumerRef.Name != machine.Name {
```

Comment:
This should be baremetalmachine

Reply:
this test case commented for now. needs to be rewritten to make sense